src/createEventEmitterProxy.test.ts

```diff
@@ -31,9 +46,13 @@ describe('createEventEmitterProxy', () => {
       sawEvent += 1;
     });
 
+    original.emit('event');
     proxy.setTarget(next);
     next.emit('event');
     next.emit('event');
+    proxy.setTarget(original);
+    next.emit('event');
+    original.emit('event');
 
     expect(sawEvent).toBe(1);
```

Comment:
in this case it's also important when the event is seen. What are your thoughts on adding another one of these expects after the `original.emit('event');` on line 49?